File: InClassCode/Week2Code/InClassLibraryExample/Main.java
public class Main {
    public static void main(String[] args) {
        // Create a library with capacity for 10 books
        Library myLibrary = new Library(10);

        System.out.println("--- Adding Books ---");
        // Create books
        Book b1 = new Book("The Way of Kings", "Brandon Sanderson");
        Book b2 = new Book("Mistborn: The Final Empire", "Brandon Sanderson");
        Book b3 = new Book("Sufficiently Advanced Magic", "Andrew Rowe");
        Book b4 = new Book("On the Shoulders of Titans", "Andrew Rowe");
        Book b5 = new Book("Unsouled", "Will Wight");
        Book b6 = new Book("Soulsmith", "Will Wight");
        Book b7 = new Book("Mother of Learning", "Domagoj Kurmaic");
        Book b8 = new Book("Iron Prince", "Bryce O'Connor");

        // Add books to library
        myLibrary.addBook(b1);
        myLibrary.addBook(b2);
        myLibrary.addBook(b3);
        myLibrary.addBook(b4);
        myLibrary.addBook(b5);
        myLibrary.addBook(b6);
        myLibrary.addBook(b7);
        myLibrary.addBook(b8);

        System.out.println("Books added to library.");

        System.out.println("\n--- Checking Out Books ---");

        // Checkout existing books
        System.out.println("Checking out 'The Way of Kings': " + myLibrary.checkedOut(b1)); // Should be true
        System.out.println("Checking out 'Unsouled': " + myLibrary.checkedOut(b5)); // Should be true

        // Try to checkout already checked out book
        System.out.println("Checking out 'Unsouled' again: " + myLibrary.checkedOut(b5)); // Should be false

        // Checkout non-existing book
        Book notInLibrary = new Book("Harry Potter", "J.K. Rowling");
        System.out.println("Checking out 'Harry Potter' (not in library): " + myLibrary.checkedOut(notInLibrary));
        // Note: Due to the bug in Library.findBook, this might incorrectly modify another book or return unexpected results
        // depending on implementation, but we are just running the client code here.

        System.out.println("\n--- Returning Books ---");

        // Return existing book
        System.out.println("Returning 'The Way of Kings': " + myLibrary.returnBook(b1)); // Should be true

        // Return book that wasn't checked out
        System.out.println("Returning 'Mistborn' (not checked out): " + myLibrary.returnBook(b2)); // Should be false

        // Return non-existing book
        System.out.println("Returning 'Harry Potter': " + myLibrary.returnBook(notInLibrary));
    }
}

File: InClassCode/Week2Code/InClassLibraryExample/main.py
from book import Book
from library import Library

def main():
    # Create a library with capacity for 10 books
    my_library = Library(10)

    print("--- Adding Books ---")
    # Create books
    b1 = Book("The Way of Kings", "Brandon Sanderson")
    b2 = Book("Mistborn: The Final Empire", "Brandon Sanderson")
    b3 = Book("Sufficiently Advanced Magic", "Andrew Rowe")
    b4 = Book("On the Shoulders of Titans", "Andrew Rowe")
    b5 = Book("Unsouled", "Will Wight")
    b6 = Book("Soulsmith", "Will Wight")
    b7 = Book("Mother of Learning", "Domagoj Kurmaic")
    b8 = Book("Iron Prince", "Bryce O'Connor")

    # Add books to library
    my_library.add_book(b1)
    my_library.add_book(b2)
    my_library.add_book(b3)
    my_library.add_book(b4)
    my_library.add_book(b5)
    my_library.add_book(b6)
    my_library.add_book(b7)
    my_library.add_book(b8)

    print("Books added to library.")

    print("\n--- Checking Out Books ---")

    # Checkout existing books
    print(f"Checking out 'The Way of Kings': {my_library.checked_out(b1)}") # Should be True
    print(f"Checking out 'Unsouled': {my_library.checked_out(b5)}") # Should be True

    # Try to checkout already checked out book
    print(f"Checking out 'Unsouled' again: {my_library.checked_out(b5)}") # Should be False

    # Checkout non-existing book
    not_in_library = Book("Harry Potter", "J.K. Rowling")
    print(f"Checking out 'Harry Potter' (not in library): {my_library.checked_out(not_in_library)}")
    # Note: Due to the bug in Library.findBook, this might incorrectly modify another book or return unexpected results
    # depending on implementation, as we are faithfully reproducing the Java logic.

    print("\n--- Returning Books ---")

    # Return existing book
    print(f"Returning 'The Way of Kings': {my_library.return_book(b1)}") # Should be True

    # Return book that wasn't checked out
    print(f"Returning 'Mistborn' (not checked out): {my_library.return_book(b2)}") # Should be False

    # Return non-existing book
    print(f"Returning 'Harry Potter': {my_library.return_book(not_in_library)}")

if __name__ == "__main__":
    main()
